OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -67,26 +84,77 @@ public override Activity Tick(Actor self)
 			if (IsCanceled || !target.IsValidFor(self))
 				return NextActivity;
 
+			if (firstTick)
+			{
+				isTakeOff = self.IsAtGroundLevel();
+				firstTick = false;
+
+				if (aircraft.Info.TakeoffSound != null && isTakeOff)
+				{
+					if (!aircraft.Info.VTOL)
+						isTakeOff = false;
```

Comment:
This `isTakeOff` handling is kind of crazy.  Can you please define a it with the correct value from the start (and define a `var groundLevel` to use for the other checks)?